chainio/interface.go

```diff
@@ -0,0 +1,50 @@
+package chainio
+
+// Blockbeat defines an interface that can be used by subsystems to retrieve
+// block data. It is sent by the BlockbeatDispatcher whenever a new block is
+// received. Once the subsystem finishes processing the block, it must signal
+// it by calling NotifyBlockProcessed.
+//
+// The blockchain is a state machine - whenever there's a state change, it's
+// manifested in a block. The blockbeat is a way to notify subsystems of this
+// state change, and to provide them with the data they need to process it. In
+// other words, subsystems must react to this state change and should consider
+// being driven by the blockbeat in their own state machines.
+type Blockbeat interface {
```

Comment:
> so the only info about the block carried by block beat is the height?

In this PR yes, but in the final PR it has more info https://github.com/lightningnetwork/lnd/pull/8922/commits/f95fc391a3bd08e0694b2e0b0e8755bbae112b52
Sorry if it's a bit confusing - this PR is created from check-picking commits from the final PR, so not all pieces fit together here. The ultimate plan is that we can remove `chainntfns` and use this package instead for all onchain data.

> also, should the name be BlockBeat (ie 2 words)

So I got the inspiration from `heartbeat`, guess I'm inventing words here but lmk what you think.

> is there any other expected implementation of this other than Beat?

Hopefully not🤓 Initially this is a concrete struct, but later on, it was changed because it allows unit tests to be written more easily.